ted/src/main/java/frc/robot/subsystems/LEDSubsystem.java

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+// ************************************************************
+// Bishop Blanchet Robotics
+// Home of the Cybears
+// FRC - Crescendo - 2024
+// File: LEDSubsystem.java
+// Intent: Forms a subsystem to control the LEDs
+// ************************************************************
+
+
+// ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ
+
+package frc.robot.subsystems;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BooleanSupplier;
+import java.util.Iterator;
+
+import frc.robot.common.LEDStateAction;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.common.LEDState;
+import com.ctre.phoenix.led.*;
+import com.ctre.phoenix.led.CANdle.LEDStripType;
+import com.ctre.phoenix.led.CANdle.VBatOutputMode;
+
+public class LEDSubsystem extends SubsystemBase {
+    private final CANdle leds;
+    
+      private final int BUFFER_LENGTH = 144;
+      private int startIdx = 0;
+      private HashMap <LEDState, LEDStateAction> ledStateActions = new HashMap<LEDState, LEDStateAction>();
+      private int blinkPeriodInHertz = 25; // increase this for longer blinks!!!
+      private int blinkCounter = 0;
+      private boolean currentBlinkState = false;
+      private boolean lastBlinkState = false;
+      private LEDState currentLEDState = LEDState.Off;
+
+
+      /**
+       * LEDSubsystem
+       * @param canID can id of the CANdle
+        */
+      public LEDSubsystem(int canID) {
+        this.leds = new CANdle(canID); // initialization of the AdressableLED
+        CANdleConfiguration configAll = new CANdleConfiguration();
+        configAll.stripType = LEDStripType.RGB;
+        configAll.brightnessScalar = 0.5;
+        configAll.vBatOutputMode = VBatOutputMode.Modulated;
+        this.leds.configAllSettings(configAll, 100);
+      }
+
+    public void configLedType(LEDStripType type) { 
+        leds.configLEDType(type, 0); 
+    }
+
+    public void configStatusLedBehavior(boolean offWhenActive) { 
```

Comment:
I don't think it makes sense for this to be a method on this class. I think we would only set this behavior at the start of time, not dynamically. You could either hardcode it to be off in this class, or you could put a variable in Constants.java and set it to that value in the constructor.

Reply:
done